index.go

```diff
@@ -24,10 +24,15 @@ type Index struct {
 // IndexChunk is a table entry in an index file containing the chunk ID (SHA256)
 // Similar to an FormatTableItem but with Start and Size instead of just offset to
 // make it easier to use throughout the application.
+// Additionally, the Validated boolean is used to mark the chunks that have been
+// checked against the hash of the underlying data. If the hash doesn't match,
+// IsInvalid will be set to true.
 type IndexChunk struct {
-	ID    ChunkID
-	Start uint64
-	Size  uint64
+	ID        ChunkID
+	Start     uint64
+	Size      uint64
+	Validated bool
+	IsInvalid bool
```

Comment:
I don't think these two are required, are they? Those should only be needed on the seed, not the chunk-level

Reply:
Hmm, I see. `IndexChunk` is a fundamental data structure that is used to describe chunks, and not related to seeds. I believe it's the wrong place to store these flags and could lead to all kinds of problems (including race conditions). It'd be better to store this somewhere else, like in the fileseed itself, or remove that for now to keep it simple. That would then require a re-validation if the same chunks are loaded again but that's basically what happened before too, so not slower than previously.
If you store it in the fileseed, basically wrapping the IndexChunk, you'll need to deal with concurrency and avoid a data race.

Reply:
Done in the latest revision.